nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -40,33 +42,38 @@ let
       empty_file=$(${pkgs.coreutils}/bin/mktemp)
 
       ${concatStrings (mapAttrsToList (n: v: ''
-        ${pkgs.coreutils}/bin/install -Dp "${v}" "${efi.efiSysMountPoint}/"${escapeShellArg n}
-        ${pkgs.coreutils}/bin/install -D $empty_file "${efi.efiSysMountPoint}/efi/nixos/.extra-files/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -Dp "${v}" "${mountPoint}/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -D $empty_file "${mountPoint}/efi/nixos/.extra-files/"${escapeShellArg n}
       '') cfg.extraFiles)}
 
       ${concatStrings (mapAttrsToList (n: v: ''
-        ${pkgs.coreutils}/bin/install -Dp "${pkgs.writeText n v}" "${efi.efiSysMountPoint}/loader/entries/"${escapeShellArg n}
-        ${pkgs.coreutils}/bin/install -D $empty_file "${efi.efiSysMountPoint}/efi/nixos/.extra-files/loader/entries/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -Dp "${pkgs.writeText n v}" "${mountPoint}/loader/entries/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -D $empty_file "${mountPoint}/efi/nixos/.extra-files/loader/entries/"${escapeShellArg n}
       '') cfg.extraEntries)}
     '';
   };
 
-  checkedSystemdBootBuilder = pkgs.runCommand "systemd-boot" {
+  checkedSystemdBootBuilder = mountPoint: pkgs.runCommand "systemd-boot" {
     nativeBuildInputs = [ pkgs.mypy python3 ];
   } ''
-    install -m755 ${systemdBootBuilder} $out
+    install -m755 ${systemdBootBuilder mountPoint} $out
     mypy \
       --no-implicit-optional \
       --disallow-untyped-calls \
       --disallow-untyped-defs \
       $out
   '';
 
-  finalSystemdBootBuilder = pkgs.writeScript "install-systemd-boot.sh" ''
-    #!${pkgs.runtimeShell}
-    ${checkedSystemdBootBuilder} "$@"
-    ${cfg.extraInstallCommands}
-  '';
+  finalSystemdBootBuilder = let
+    installDirs =
+      if cfg.mirroredBoots != []
+      then cfg.mirroredBoots
+      else [efi.efiSysMountPoint];
+  in
+    pkgs.writeShellScript "install-systemd-boot.sh"
+      (lib.concatMapStrings (x: "${checkedSystemdBootBuilder x} \"$@\"\n") installDirs)
```

Comment:
I think that, like in [grub.nix](https://github.com/NixOS/nixpkgs/blob/ec6cc9c826e08ec1aa4212c40ad92eb5186a9053/nixos/modules/system/boot/loader/grub/grub.nix#L734), the usual error detection [`set -e`](https://stackoverflow.com/questions/19622198/what-does-set-e-mean-in-a-bash-script) should be activated, and maybe also [`set -u`](https://unix.stackexchange.com/questions/544993/what-does-set-u-do-in-bash-script).

Why keeping on passing `$@`? I don't see any argument neither used, [nor passed](https://github.com/NixOS/nixpkgs/blob/19cbff58383a4ae384dea4d1d0c823d72b49d614/nixos/modules/system/activation/switch-to-configuration.pl#L102) (so far).

Nitpick: here each mountpoint generates a new derivation, the mountpoint could instead be passed as an envvar `mountpoint=${escapeShellArg x} ${checkedSystemdBootBuilder}`. Again that's likely just a few derivations in practice, so just a nitpick.